appinventor/components/src/com/google/appinventor/components/runtime/SpeechRecognizer.java

```diff
@@ -110,4 +127,59 @@ public void AfterGettingText(String result) {
     EventDispatcher.dispatchEvent(this, "AfterGettingText", result);
   }
 
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onPartialResult(String text) {
+    result = text;
+    AfterGettingText(result);
+  }
+
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onResult(String text) {
+    result = text;
+    AfterGettingText(result);
+    onDelete();
+  }
+
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onError(String message) {
+    result = message;
+    AfterGettingText(result);
+  }
+
+  @Override
+  public void onDelete() {
+    speechRecognizerController = null;
+    recognizerIntent = null;
+  }
+
+  @SimpleProperty(
+      category = PropertyCategory.BEHAVIOR,
+      description = "If set, an app can retain their older behaviour.")
+  public boolean UseLegacy() {
+      return useLegacy;
+  }
+
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN,
+      defaultValue = "True")
+  @SimpleProperty(description = "Used to set UseLegacy property based on user's choice.")
```

Comment:
This description seems like a tautology to me. Instead, you should include a more user-friendly description, e.g. "If set to true, a separate dialog is used to recognize speech. If false, speech is recognized without changing the user interface. When false, the SpeechRecognizer also provides partial results."